scheduler/node/node.go

```diff
@@ -81,3 +89,32 @@ func (n *Node) AddContainer(container *cluster.Container) error {
 	n.Containers = append(n.Containers, container)
 	return nil
 }
+
+// ReserveResource reserve some resources for a container
+func (n *Node) ReserveResource(config *cluster.ContainerConfig) error {
+	n.Lock()
+	defer n.Unlock()
+	if config != nil {
+		memory := config.Memory
+		cpus := config.CpuShares
+		if (n.TotalMemory-n.UsedMemory)-memory < 0 || (n.TotalCpus-n.UsedCpus)-cpus < 0 {
```

Comment:
I'm wondering if you are going to run into precision problems here, as the check and the addition lower down there might not end up with the same answer for cpus.